src/interface/sparse_pack_base.cpp

```diff
@@ -122,12 +123,19 @@ SparsePackBase SparsePackBase::Build(T *pmd, const PackDescriptor &desc) {
   auto coords_h = Kokkos::create_mirror_view(pack.coords_);
 
   // Fill the views
-  ForEachBlock(pmd, [&](int b, mbd_t *pmbd) {
-    int idx = 0;
-    coords_h(b) = pmbd->GetBlockPointer()->coords_device;
+  int idx = 0;
+  ForEachBlock(pmd, [&](int block, mbd_t *pmbd) {
+    int b = 0;
+    if (!desc.flat) {
+      idx = 0;
+      b = block;
+      coords_h(b) = pmbd->GetBlockPointer()->coords_device;
+    }
 
     for (int i = 0; i < nvar; ++i) {
-      pack.bounds_h_(0, b, i) = idx;
+      if (!desc.flat) {
+        pack.bounds_h_(0, b, i) = idx;
+      }
```

Comment:
Can't we still set this with the actual block index, i.e. `pack.bounds_h_(0, block, i) = idx`? I am not sure what the use case would be, but it would allow all of the `Get*Bound` stuff to work returning the flat index.

Reply:
Is this actually desired behaviour though? I wonder if `GetBounds` should return 0, and nvars always in the flat case? I could add an if statement to that effect maybe?

Reply:
Now updated `GetLowerBound` returns 0 and `GetUpperBound` returns total numbers of vars and blocks (i.e., pack size) for flat packs.